e2e_tests/tests/all_providers/mod.rs

```diff
@@ -113,3 +114,31 @@ fn sign_verify_with_provider_discovery() -> Result<()> {
     let key_name = String::from("sign_verify_with_provider_discovery");
     client.generate_rsa_sign_key(key_name)
 }
+
+#[test]
+fn list_keys() {
```

Comment:
Also, shouldn't this be run for each provider, just in case?

Reply:
My idea was that `list_keys` should return keys from all providers, that's why I put it there. I should test that by creating a key in all providers!

Reply:
Ah, yes, that makes more sense!